### ISSUE TITLE
sudo is not nessisary.

### ISSUE THREAD

**Issue by mrshannon** [mafm/zfs-snapshot-disk-usage-matrix#1]
The `sudo` command is not necessary and is generally considered bad practice in a script.  The script will run fine without it because ZFS allows anyone to use non modifying commands.

**Comment by mafm** [comment]
Thanks for the feedback.

I take your point, but not sure you're entirely correct.

Unless the linux zfs packages have changed recently you need root to run any zfs command. On solaris, and (iirc) bsd you don't need root for non-modifying zfs commands.

What people do in practice on linux is put the non-modifying zfs commands in the sudoers file for passwordless sudo, so they're available to everyone.

There should be a note about this in the readme, but since you're flagging this I probably forgot to mention it.

Basically agree with you, it's clearly not ideal. But not sure I see an alternative. Maybe the script should not do it unless you explicitly ask for it, but that doesn't seem ideal either.

Open to suggestions if you have a better idea.

**Comment by mrshannon** [comment]
I am running zfsonlinux 0.7.5 and even a user in no groups at all can run non-modifying zfs commands.  This includes `destroy` with a dry run as used in your script.  I did verify that nothing related was installed into the sudoes file (or sudoers.d).

If root privileges were truly needed then it would be better to bail out and ask the user to run the script as root.  In which case they can invoke `sudo` themselves.

Calling `sudo` inside a script just seems shady to me.  But if you still think that it is the best way for your program may I suggest adding the `-k` flag, it will force `sudo` to ask for a password again (even if the timeout has not expired).  I don't typically mind if a program asks me to upgrade it's privileges, but it always unnerves me when it upgrades it's privileges without me knowing it.

Excellent script by the way, I have been looking for a way to get correct snapshot size information for years.

**Comment by seonwoolee** [comment]
I concur with @mrshannon: I do not need sudo to run non-modifying zfs commands. I've also verified that I have not modified the sudoers file to this effect.

Furthermore, it's actually possible to use ZFS delegation to allow non root users to run zfs modifying commands by using zfs allow (which must be run as root, obviously)

**Comment by mailinglists35** [comment]
please update https://github.com/mafm/zfs-snapshot-disk-usage-matrix/blob/master/README.md as well. it still says sudo is needed

**Comment by mafm** [comment]
> please update https://github.com/mafm/zfs-snapshot-disk-usage-matrix/blob/master/README.md as well. it still says sudo is needed

Done just now.
